Utilities/StorageFactory/README.md

```diff
@@ -27,16 +27,59 @@ A singleton used to aggragate statistics about all storage calls for each protoc
 ### `edm::storage::StorageAccount::StorageClassToken`
 Each protocol is associated to a token for quick lookup.
 
+
+## Generic storage proxies
+
+This facility resembles the `edm::storage::LocalCacheFile` and `edm::storage::StorageAccountProxy` in the way that `edm::storage::Storage` objects constructed by the concrete `edm::storage::StorageMaker` are wrapped into other `edm::storage::Storage` objects.
+
+The proxies are configured via `TFileAdaptor`'s `storageProxies` `VPSet` configuration parameter. The proxies are wrapped in the order they are specified in the `VPSet`, i.e. the first element wraps the concrete `edm::storage::Storage`, second element wraps the first element etc. The `edm::storage::StorageAccountProxy` and `edm::storage::LocalCacheFile` wrap the last storage proxy according to their usual behavior.
+
+Each concrete proxy comes with two classes, the proxy class itself (inheriting from the `edm::storage::StorageProxyBase`) and a maker class (inheriting from the `edm::storage::StorageProxyMaker`). This "factory of factories" pattern is used because a maker is created once per job (in `TFileAdaptor`), and the maker object is used to create a proxy object for each file.
+
+### Concrete proxy classes
+
+The convention is to use the proxy class name as the plugin name for the maker, as the proxy is really what the user would care for. The headings of the subsections correspond tothe plugin names.
+
+#### `StorageTracerProxy`
+
+The `edm::storage::StorageTracerProxy` (and the corresponding `edm::storage::StorageTracerProxyMaker`) produce a text file with a trace of all IO operations at the `StorageFactory` level. The behavior of each concrete `Storage` object (such as further splitting of read requests in `XrdAdaptor`) is not captured in these tracers. The structure of the trace file is described in a preamble in the trace file.
+
+The plugin as a configuration parameter for a pattern for the trace files. The pattern must contain at least one `%I`. The maker has an atomic counter for the files, and all occurrences of `%I` are replaced with the value of that counter for the given file.
+
+There is an `edmStorageTracer.py` script for doing some analyses of the traces.
+
+#### `StorageAddLatencyProxy`
+
+The `edm::storage::StorageAddLatencyProxy` (and the corresponding `edm::storage::StorageAddLatencyProxyMaker`) can be used to add artifical latency to the IO operations. The plugin has configuration parameters for latencies of singular reads, vector reads, singular writes, and vector writes.
+
+If used together with `StorageTracerProxy` to e.g. simulate the behavior of high-latency storage systems with e.g. local files, the `storageProxies` `VPSet` should have `StorageAddLatencyProxy` first, followed by `StorageTracerProxy`.
```

Comment:
Didn't you also have to set some other parameter somewhere to fake a local file to seem like it is a remote file? If so, maybe mention that here?

Reply:
Described now in XrdAdaptor's README. In the future I plan to describe the cache hints here (or in `IOPool/TFileAdaptor`?) as well (basically moving the documentation from https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideTFileAdaptor)